File: Java/프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        // k진법 변환
        String num = "";
        while (n > 0) {
            num += n % k;
            n /= k;
        }
        // 역순으로 뒤집어서 변환
        String number = new StringBuilder(num).reverse().toString();
        
        // 0을 기준으로 숫자 나누기
        String[] parts = number.split("0");
        
        int count = 0;
        // 각 숫자에 대해 소수 여부를 확인
        for (String part : parts) {
            if (!part.equals("") && isPrime(Long.parseLong(part))) {
                count++;
            }
        }
        
        return count;
    }
    
    // 소수 판별 함수
    public boolean isPrime(long num) {
        // 1은 소수가 아니다.
        if (num < 2) {
            return false;
        }
        for (long i = 2; i * i <= num; i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
}

File: Python3/프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):  # 2부터 √n까지 나누어 떨어지는지 확인
        if n % i == 0:
            return False
    return True

def convert_to_base(n, base):
    T = "0123456789ABCDEF"
    if n == 0:
        return "0"
    
    result = ''
    while n > 0:
        result = T[n % base] + result  # 나머지를 앞에 추가
        n //= base  # 몫을 다음 숫자로 설정
    
    return result

def solution(n, k):
    answer = 0
    
    num = convert_to_base(n, k)
    nums = num.split('0')
    
    for word in nums:
        if word != '' and is_prime(int(word)):
            answer += 1
    return answer